mne/io/edf/edf.py

```diff
@@ -425,10 +425,6 @@ def _get_info(fname, stim_channel, annot, annotmap, eog, misc, exclude,
         cals[idx] = 1
     if 'stim_data' in edf_info and stim_channel == 'auto':  # For GDF events.
         cals = np.append(cals, 1)
-    # Check that stimulus channel exists in dataset
-    if not ('stim_data' in edf_info or
-            'EDF Annotations' in ch_names) and stim_channel == 'auto':
-        stim_channel = None
```

Comment:
this will change the behavior for EDF+ files with annotation channel. any argument why this is a bug fix?

Reply:
EDF+ files with annotation channel are not affected, because `'EDF Annotations' in ch_names` is `True`, therefore `not (False or True)` is `False`, therefore `False and whatever` is `False`.